### PR TITLE
Use futex.rs for Windows thread parking

### PR DESCRIPTION
**Pull request by ChrisDenton**
If I'm not overlooking anything then the Windows 10+ thread parking implementation is practically the same as the futex.rs implementation. So we may as well use the same implementation for both. The old version is still kept around for Windows 7 support.

r? @joboet if you wouldn't mind double checking I've not missed something

### CODE REVIEW

File: library/std/src/sys/sync/thread_parking/futex.rs [hunk 1]
```diff
@@ -1,15 +1,18 @@
+#![deny(unsafe_op_in_unsafe_fn)]
```
Comment (joboet) [inline library/std/src/sys/sync/thread_parking/futex.rs:1]: Please use forbid to deny this once and for all.
```suggestion
#![forbid(unsafe_op_in_unsafe_fn)]
```

**Review by joboet — APPROVED** [review]
That's much better, thank you!
r=me with the nit applied and once CI passes

**Comment by workingjubilee** [comment]
If this fails in rollups, then #127813 should be unblocked and queued first.

**Pull request merged**